spec/unit/pops/loaders/loaders_spec.rb

```diff
@@ -234,15 +234,15 @@ def compile_and_get_notifications(code)
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when dependency is present in metadata.json" do
-      File.stubs(:read).with(user_metadata_path).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
+      Puppet::FileSystem.stubs(:read).with(user_metadata_path, {:encoding => 'utf-8'}).returns user_metadata.merge('dependencies' => [ { 'name' => 'test-usee'} ]).to_pson
       expect(eval_and_collect_notices(<<-CODE, node)).to eq(['ok'])
         assert_type(User::WithUseeZero, [0])
         notice(ok)
       CODE
     end
 
     it "an autoloaded type can reference an autoloaded type alias from another module when other module has it declared in init.pp" do
```

Comment:
The mocking here is a bit fragile, as we're trying to bring more of Puppet under using the FileSystem abstraction we are finding it doesn't work out well with this spec due to mocking not really working for something that tests multiple parts of Puppet and not a singular unit. Some/all of these specs in this class should likely be moved over to integration and the mocking bits removed.